x-pack/plugins/ml/public/application/components/field_stats_flyout/populated_fields/populated_fields_cache_manager.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+type StringifiedQueryKey = string;
+type UpdatedTimestamp = number;
+
+const DEFAULT_EXPIRATION_MS = 60000;
+export class PopulatedFieldsCacheManager {
+  private _expirationDurationMs = DEFAULT_EXPIRATION_MS; // duration in ms
+
+  private _resultsCache = new Map<StringifiedQueryKey, any>();
+  _lastUpdatedTimestamps = new Map<StringifiedQueryKey, UpdatedTimestamp>();
```

Comment:
should it be `private readonly`?

Reply:
Updated here [`2dcb364` (#163371)](https://github.com/elastic/kibana/pull/163371/commits/2dcb3644dc0702612c4d46b5e44523f04a1c865b)